src/main/java/cz/muni/fi/pa165/library/services/MemberService.java

```diff
@@ -0,0 +1,115 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Katarína Hermanová
+ * UČO 433511
+ * Github katHermanova
+ */
+@Service
+public class MemberService {
+
+    @Autowired
+    private MemberRepository memberRepository;
+    private static final Logger LOGGER = LoggerFactory.getLogger(MemberService.class);
+
+    public MemberService(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    public Member findById(long id) {
+        if (id < 0) {
+            throw new IllegalArgumentException("ID can not be less than 0.");
+        }
+        if (memberRepository.findById(id).isPresent()) {
+            return memberRepository.findById(id).get();
+        }
+        return null;
+    }
+
+    public List<Member> findByFirstName(String firstName) {
+        if (firstName == null || firstName.isEmpty()) {
+            throw new IllegalArgumentException("FirstName can not be empty.");
+        }
+        List<Member> foundMembers = new ArrayList<>();
+
+        for (Member member: memberRepository.findAll()) {
+            if (member.getFirstName().equals(firstName)) {
+                foundMembers.add(member);
+            }
+        }
+        return foundMembers;
+    }
+
+    public List<Member> findBySurname(String surname) {
+        if (surname == null || surname.isEmpty()) {
+            throw new IllegalArgumentException("Surname can not be empty.");
+        }
+        List<Member> foundMembers = new ArrayList<>();
+
+        for (Member member: memberRepository.findAll()) {
+            if (member.getSurname().equals(surname)) {
+                foundMembers.add(member);
+            }
+        }
+        return foundMembers;
+    }
+
+    public List<Member> findLibrarians() {
+        return getMembersByLibraryRole(true);
+    }
+
+    public List<Member> findNonLibrarians() {
+        return getMembersByLibraryRole(false);
+    }
+
+    public List<Member> findAll() {
+        return (List<Member>)memberRepository.findAll();
+    }
+
+    public void addMember(Member member) {
+        if (member == null) {
+            throw new IllegalArgumentException("Can not add non-existing member.");
+        }
+        memberRepository.save(member);
+    }
+
+    public void deleteMember(long id) {
+        if (id < 0) {
+            throw new IllegalArgumentException("ID can not be less than 0.");
+        }
+        if (memberRepository.existsById(id)) {
+            memberRepository.deleteById(id);
+        }
+    }
+
+    public void deleteAllMembers() {
+        memberRepository.deleteAll();
+    }
+
+    public long count() {
+        return memberRepository.count();
+    }
+
+    private List<Member> getMembersByLibraryRole(boolean lookingLibrarians) {
```

Comment:
add two different methods instead, one to getAllMembers, second to getOnlyLibrarians

Reply:
No problem with that but is it necessary? Because I have 3 methods to get members...findLibrarians(), findNonLibrarians() and findAll()...but findLibrarians() and findNonLibrarians() are using this method (getMembersByLibraryRole()) because I do not wanna have the same code there

Reply:
If there were many roles as an enum for example, like Librarian, Staff, Cook, Citize, then it would make sense to have a method, which would take this enum as a parameter and filter out only desired members. However, this pattern, using flag parameter is suboptimal. You duplicate the code anyway, inside the method. Please create two separate methods